js/src/run_trees.ts

```diff
@@ -246,13 +246,30 @@ export class RunTree implements BaseRun {
     };
     config.extra = { ...config.extra, metadata: dedupedMetadata };
     Object.assign(this, { ...defaultConfig, ...config, client });
+
+    // If start_time was provided but id was not, regenerate id from start_time
+    if (config.start_time !== undefined && !config.id) {
+      this.id = uuidFromTime(config.start_time);
```

Comment:
There's new hard requirement that the run id must be generated from start time then right?